pandas/tests/frame/test_constructors.py

```diff
@@ -528,6 +528,29 @@ def test_constructor_dict_of_tuples(self):
         expected = DataFrame({k: list(v) for k, v in data.items()})
         tm.assert_frame_equal(result, expected, check_dtype=False)
 
+    def test_constructor_dict_of_ranges(self):
+        data = {'a': range(3), 'b': range(3, 6)}
```

Comment:
might be worth using `range(2, -1, -1)` for b so that the expected is the same for all three tests.

Reply:
Hi,

I did it this way to by copying ``test_constructor_dict_of_tuples`` just above. Probably not worth it to align all of these styles, though in principle it would be clearer IMO.

Reply:
the reason i mentioned it is that it would make parametrization easier. either in this PR or a follow-on. you could potentially then parametrize `a` and `b` independently to create the cartesian product (not sure if this would work simply with generators or whether they would be exhausted) and then add in list, tuple, series and index for completeness. this would ensure that the independence of the column parsing for each item in the dictionary.

Reply:
Agree this would be a good issue for a follow-up. This module looks like it is from before Pandas switched to pytest, so parametrization could make things clearer.